extensions/interactions/NumberWithUnits/directives/number-with-units-validation.service.ts

```diff
@@ -16,130 +16,144 @@
  * @fileoverview Validator service for the number with units interaction.
  */
 
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 // TODO(#7403): Convert this to partial imports.
 import math from 'mathjs';
 
-require('domain/objects/NumberWithUnitsObjectFactory.ts');
-require('interactions/base-interaction-validation.service.ts');
-
-angular.module('oppia').factory('NumberWithUnitsValidationService', [
-  'NumberWithUnitsObjectFactory', 'baseInteractionValidationService',
-  'WARNING_TYPES',
-  function(NumberWithUnitsObjectFactory, baseInteractionValidationService,
-      WARNING_TYPES) {
-    return {
-      getCustomizationArgsWarnings: function(customizationArgs) {
-        return [];
-      },
-      getAllWarnings: function(
-          stateName, customizationArgs, answerGroups, defaultOutcome) {
-        var warningsList = [];
-
-        warningsList = warningsList.concat(
-          this.getCustomizationArgsWarnings(customizationArgs));
-
-        try {
-          NumberWithUnitsObjectFactory.createCurrencyUnits();
-        } catch (parsingError) {}
-
-        var checkEquality = function(earlierRule, laterRule) {
-          var answer = NumberWithUnitsObjectFactory.fromDict(
-            earlierRule.inputs.f);
-          var inputs = NumberWithUnitsObjectFactory.fromDict(
-            laterRule.inputs.f);
-
-          var answerString = answer.toMathjsCompatibleString();
-          var inputsString = inputs.toMathjsCompatibleString();
-
-          var answerList = NumberWithUnitsObjectFactory.fromRawInputString(
-            answerString).toDict();
-          var inputsList = NumberWithUnitsObjectFactory.fromRawInputString(
-            inputsString).toDict();
-          return angular.equals(answerList, inputsList);
-        };
+import { AppConstants } from 'app.constants';
+import { baseInteractionValidationService } from
+  'interactions/base-interaction-validation.service.ts';
+import { NumberWithUnitsObjectFactory } from
+  'domain/objects/NumberWithUnitsObjectFactory.ts';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NumberWithUnitsValidationService {
+  constructor(
+    private nwuof: NumberWithUnitsObjectFactory,
```

Comment:
Should we name the variables like this? Even if we want to have shorter ones, I would still be against using shortcuts, maybe something like `unitNumbersFactory`?

Reply:
Done